superset-frontend/src/datasource/ChangeDatasourceModal.tsx

```diff
@@ -171,7 +171,7 @@ const ChangeDatasourceModal: FunctionComponent<ChangeDatasourceModalProps> = ({
 
   const handleChangeConfirm = () => {
     SupersetClient.get({
-      endpoint: `/datasource/get/${confirmedDataset?.type}/${confirmedDataset?.id}`,
+      endpoint: `/datasource/get/${confirmedDataset?.type}/${confirmedDataset?.id}/`,
```

Comment:
Bycatch - I noticed that the datasource request was always redirected to the one with the trailing slash.